src/main/scala/chisel3/util/experimental/decode/TruthTable.scala

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.experimental.decode
+
+import chisel3.util.BitPat
+
+sealed class TruthTable(val table: Map[BitPat, BitPat], val default: BitPat) {
+
+  def inputWidth = table.head._1.getWidth
+
+  def outputWidth = table.head._2.getWidth
+
+  override def toString: String = {
+    def writeRow(map: (BitPat, BitPat)): String =
+      s"${TruthTable.bpStr(map._1)}->${TruthTable.bpStr(map._2)}"
+
+    (table.map(writeRow) ++ Seq(s"${" "*(inputWidth + 2)}${TruthTable.bpStr(default)}")).toSeq.sorted.mkString("\n")
+  }
+
+  def copy(table: Map[BitPat, BitPat] = this.table, default: BitPat = this.default) = new TruthTable(table, default)
+
+  override def equals(y: Any): Boolean = {
+    y match {
+      case y: TruthTable => toString == y.toString
+      case _ => false
+    }
+  }
```

Comment:
I remember I used SortedMap before , and failed in a 2.12 but successes in 2.13, I will retry this.

Reply:
I just take 2 hours to try this.
I found the main problem is we cannot compare two `SortedMap[BitPat, BitPat]`. since `BitPat` in `SortedMap` is a reference.

Reply:
So I personally prefer using String comparing here. I use sorted the serialized string, which provide a safer way to implement this.

Reply:
Yeah it's a bit awkward, but that derives from the referential equality of `BitPat` so I think this is the best solution.